examples/waspleau/Notes.md

```diff
@@ -0,0 +1,198 @@
+# Observations
+
+## 1) Loading hard coded data from query (but no entities) causes Prisma error
+
+I was attempting to return hard coded JSON from a query without any entities defined, but ran into the following issue on `wasp start`:
+
+```
+Server: /Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/node_modules/.prisma/client/index.js:3
+Server (stderr):     throw new Error(
+Server (stderr):           ^
+Server (stderr):
+Server (stderr): Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
+Server (stderr): In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
+Server (stderr):     at new PrismaClient (/Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/node_modules/.prisma/client/index.js:3:11)
+Server (stderr):     at createDbClient (file:///Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/src/dbClient.js:5:24)
+Server (stderr):     at file:///Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/src/dbClient.js:11:18
+Server (stderr):     at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
+Server (stderr):     at async Loader.import (internal/modules/esm/loader.js:166:24)
+Server (stderr):     at async Object.loadESM (internal/process/esm_loader.js:68:5)
+Server (stderr): [nodemon] app crashed - waiting for file changes before starting...
+```
+
+Therefore, I assumed I needed to migrate, however...
+
+### Running `wasp db migrate-dev` without any entities defined
+
+```
+Database successfully set up!
+Copying migrations folder from Wasp to Prisma project...
+Done copying migrations folder.
+Performing migration...
+Prisma schema loaded from ../db/schema.prisma
+Datasource "db": SQLite database "dev.db" at "file:./dev.db"
+
+Already in sync, no schema change or pending migration was found.
+
+Running generate... (Use --skip-generate to skip the generators)
+Error:
+You don't have any models defined in your schema.prisma, so nothing will be generated.
+You can define a model like this:
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+
+More information in our documentation:
+https://pris.ly/d/prisma-schema
+
+Error: Migrate dev failed: Migrate (dev) failed with exit code: 1
+```
+
+The fix was to add a dummy migration just to get past the Prisma error.
+
+
+| Bug captured here: https://github.com/wasp-lang/wasp/issues/398 |
+| --------------------------------------------------------------- |
+
+## 2) Axios version pinning
+
+```
+wasp: Error: Dependency conflict for user npm dependency (axios, ^0.21.1): Version must be set to the exactly the same version as the one wasp is using: ^0.21.1
```

Comment:
How did you fix this one? From what I see you did use the exact same version as Wasp does, but Wasp still wasn't happy?

Reply:
This error was before that and it eventually worked. I can update the note to that effect. This was if I added the axios dependency without a version, it complained, so I had to pin to what Wasp used and then it went away.

Reply:
Updated here: https://github.com/wasp-lang/wasp/pull/412/commits/69fd53485f8f520c7c8c4066557e420271bfebd4